src/olympia/files/utils.py

```diff
@@ -229,13 +226,16 @@ def type(self):
             else amo.ADDON_EXTENSION
         )
 
-    @property
-    def strict_max_version(self):
-        return get_simple_version(self.gecko.get('strict_max_version'))
-
-    @property
-    def strict_min_version(self):
-        return get_simple_version(self.gecko.get('strict_min_version'))
+    def get_strict_version_for(self, *, key, application):
+        strict_key = f'strict_{key}_version'
+        if application == amo.FIREFOX:
+            return get_simple_version(self.gecko.get(strict_key))
+        elif application == amo.ANDROID:
+            return get_simple_version(
+                self.gecko_android.get(strict_key, self.gecko.get(strict_key))
+            )
+        else:
+            return get_simple_version(None)
```

Comment:
I found this function a little difficult to understand with the multiple layers of `get`.  Do you think something more like the following works? 
```suggestion
    def get_strict_version_for(self, *, key, application):
        strict_key = f'strict_{key}_version'
        if application == amo.FIREFOX:
            strict_version_string = self.gecko.get(strict_key)
        elif application == amo.ANDROID:
            strict_version_string = self.gecko_android.get(strict_key, self.gecko.get(strict_key))
        else:
            strict_version_string = None
        return get_simple_version(strict_version_string)
```
(might need reformatting.  I looked for a way to avoid doing `self.gecko.get(strict_key)` twice but there wasn't an obvious way to avoid it)